### ISSUE TITLE
Ratings data `type` doesn't match Google Places supported types

### ISSUE THREAD

**Issue by semionaut** [juanitacosmica/FinalProject_RoadTrippr#28]
As we know `type` is a supported field in the Places API. Not all the types reflected in the ratings data match the [Google Places supported types](https://developers.google.com/maps/documentation/places/web-service/search-nearby#table1). In Place Search, the specified `type` needs to be an exact match.

Places API also supports [additional place types](https://developers.google.com/maps/documentation/places/web-service/search-nearby#table2).

Another limitation, only one `type` may be specified per Places search.

Review of `types` in our dataset:
```
art-galleries
bakeries
beaches
beauty-spas
burger-pizza
cafes
churches
dance-clubs
gardens
gyms
juice-bars
malls
monuments
museums
parks
pubs-bars
resorts
restaurants
swimming-pools
theatres
view-points
zoo
```

Corresponding supported `type`s we need to match:
| Ratings Dataset | Places Type | Match (y/n) |
| ----------------- |:-------------:| :-------------:|
| art-galleries | `art_gallery` | N |
| bakeries | `bakery` | N |
| beaches | (no type) | N |
| beauty-spas | `beauty_salon`, `hair_care`, `spa` | N |
| burger-pizza | (no type) | N |
| cafes | `cafe` | N |
| churches | `church`, `hindu_temple`, `mosque`, `place_of_worship`, `synagogue` | N |
| dance-clubs | `night_club` | N |
| gardens | (no type) | N |
| gyms | `gym` | N |
| juice-bars | (no type) | N |
| malls | `shopping_mall` | N |
| monuments | `landmark` | N |
| museums | `museum` | N |
| parks | `park` | N |
| pubs-bars | `bar`  | N |
| resorts | `amusement_park` | N |
| restaurants | `restaurant` | N |
| swimming-pools | (no type) | N |
| theatres | `movie_theater`  | N |
| view-points | `natural_feature`(?), `landmark`(?) | N |
| zoo | `zoo` | Y |

Where no analogous place type exists, we can try passing these to place search [by keyword](https://developers.google.com/maps/documentation/places/web-service/search-nearby#keyword). However the API documentation warns: 
> Explicitly including location information using this parameter may conflict with the location, radius, and rankby parameters, causing unexpected results. 

As another alternative, we can explore pushing this to keyword search __only__ instead of using the `type` parameter.

**Comment by semionaut** [comment]
I applied a quick fix here: 3c4a04938b509ca1474fce0e980f3bae7bf2fe60 to allow me to continue working on the API code.

Note some choices were made. Where no corresponding `type` exists, we can try to pass those as `keyword` instead of `type` using a conditional.

Later on, imo this should be applied lower down as a cleaning step, prior to the PCA/ML/clustering.

**Comment by semionaut** [comment]
This is updated in 7a7efe66eaecd1b643f20996bb4693e58b812229 

The function is currently set to always use `keyword` in Nearby Search, though as Google suggests using `type` where possible is more robust and is recommended. I'll later sort out the logic of which locations need to search by `type` and which should search by `keyword`. But this is stable for now.

**Comment by semionaut** [comment]
We discussed and decided that since these are working well, to stick with just `keyword` searches. Reverted the column header cleaning slightly in 4f5d32c147e38e782305e4242fb6b5b008eb5fbf

We should keep an eye out for further issues with using keyword search. Closing this issue for now.

Later, we can revisit the logic of which categories should always search by canonical `type` and which should fall back to `keyword`.
